doc/rules/basic/numeric_literal_separator.rst

```diff
@@ -0,0 +1,86 @@
+==================================
+Rule ``numeric_literal_separator``
+==================================
+
+Adds separators to numeric literals of any kind.
+
+Configuration
+-------------
+
+``override_existing``
+~~~~~~~~~~~~~~~~~~~~~
+
+Whether literals already containing underscores should be reformatted.
+
+Allowed types: ``bool``
+
+Default value: ``false``
+
+``strategy``
+~~~~~~~~~~~~
+
+Whether numeric literal should be separated by underscores or not.
+
+Allowed values: ``'no_separator'`` and ``'use_separator'``
+
+Default value: ``'no_separator'``
```

Comment:
i'm not sure of the default.

Reply:
It was suggested by me [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6761#discussion_r1214414232) and agreed by @kubawerlos so I implemented it like this. All the years I've been programming I have never worked with codebase that used literal separators, that's why I believe it's sane default.

Reply:
well, it was not possible for long in PHP, no dounts there is not adoption.

I aim to use everywhere personally, basically a must have for me for TS or python.
if both of you think this set of defaults is good, let's keep the defaults as-is

Reply:
Must personally agree with @keradus, but I am OK with it as long as I have the option to set it for myself.